PhysicsTools/NanoAOD/python/nano_cff.py

```diff
@@ -212,7 +212,35 @@ def nanoAOD_recalibrateMETs(process,isData):
         table.variables.muonSubtrFactor = Var("1-userFloat('muonSubtrRawPt')/(pt()*jecFactor('Uncorrected'))",float,doc="1-(muon-subtracted raw pt)/(raw pt)",precision=6)
     process.metTables += process.corrT1METJetTable
 #    makePuppiesFromMiniAOD(process,True) # call this before in the global customizer otherwise it would reset photon IDs in VID
-    runMetCorAndUncFromMiniAOD(process,isData=isData,metType="Puppi",postfix="Puppi",jetFlavor="AK4PFPuppi")
+    nanoAOD_PuppiV15_switch = cms.PSet(
+            recoMetFromPFCs = cms.untracked.bool(False),
+            reclusterJets = cms.untracked.bool(False),
+            )
+    run2_nanoAOD_106Xv1.toModify(nanoAOD_PuppiV15_switch,recoMetFromPFCs=True,reclusterJets=True)
+    runMetCorAndUncFromMiniAOD(process,isData=isData,metType="Puppi",postfix="Puppi",jetFlavor="AK4PFPuppi", recoMetFromPFCs=bool(nanoAOD_PuppiV15_switch.recoMetFromPFCs), reclusterJets=bool(nanoAOD_PuppiV15_switch.reclusterJets))
```

Comment:
Studying a bit more in detail the MET recipe TWiki:
https://twiki.cern.ch/twiki/bin/view/CMS/MissingETUncertaintyPrescription#Puppi_MET

it seems that this call also needs
pfCandColl=cms.InputTag("puppiForMET")
when used with recoMetFromPFCs=True
in order to pick up the right collection for PUPPI MET (namely the puppiForMET collection which has photons vetoed and not the puppi collection.
To me it looks like one can't simultaneously put recoMetFromPFCs=True and reclusterJets=True, because they need different pfCandColl.
So one either has to recluster jets by hand and then call the function, or call the function twice somehow...not very elegant. The most elegant way may be to modify the function to allow for two different pfCandColl, one for recoMetFromPFCs and one for reclusterJets.